python/ray/tune/examples/pbt_transformers/pbt_transformers.py

```diff
@@ -119,34 +32,76 @@ def tune_transformer(num_samples=8,
 
     task_data_dir = os.path.join(data_dir, task_name.upper())
 
+    num_labels = glue_tasks_num_labels[task_name]
+
+    config = AutoConfig.from_pretrained(
+        model_name, num_labels=num_labels, finetuning_task=task_name)
+
     # Download and cache tokenizer, model, and features
     print("Downloading and caching Tokenizer")
+    tokenizer = AutoTokenizer.from_pretrained(model_name)
 
     # Triggers tokenizer download to cache
-    AutoTokenizer.from_pretrained(model_name)
     print("Downloading and caching pre-trained model")
+    AutoModelForSequenceClassification.from_pretrained(
+        model_name,
+        config=config,
+    )
 
-    # Triggers model download to cache
-    AutoModelForSequenceClassification.from_pretrained(model_name)
+    def get_model():
+        return AutoModelForSequenceClassification.from_pretrained(
+            model_name,
+            config=config,
+        )
 
     # Download data.
     download_data(task_name, data_dir)
 
-    config = {
-        "model_name": model_name,
-        "task_name": task_name,
-        "data_dir": task_data_dir,
-        "per_gpu_val_batch_size": 32,
-        "per_gpu_train_batch_size": tune.choice([16, 32, 64]),
+    data_args = GlueDataTrainingArguments(
+        task_name=task_name, data_dir=task_data_dir)
+
+    train_dataset = GlueDataset(
+        data_args, tokenizer=tokenizer, mode="train", cache_dir=task_data_dir)
+    eval_dataset = GlueDataset(
+        data_args, tokenizer=tokenizer, mode="dev", cache_dir=task_data_dir)
+
+    training_args = TrainingArguments(
+        output_dir=".",
+        learning_rate=1e-5,  # config
+        do_train=True,
+        do_eval=True,
+        evaluate_during_training=True,
+        eval_steps=(len(train_dataset) // 16) + 1
+        if not smoke_test else 1,  # config
+        save_steps=(len(train_dataset) // 16) + 1
+        if not smoke_test else 1,  # config,
+        num_train_epochs=2,  # config
+        max_steps=-1,
+        per_device_train_batch_size=16,  # config
+        per_device_eval_batch_size=16,  # config
+        warmup_steps=0,
+        weight_decay=0.1,  # config
+        logging_dir="./logs",
+    )
+
+    trainer = Trainer(
+        model_init=get_model,
+        args=training_args,
+        train_dataset=train_dataset,
+        eval_dataset=eval_dataset,
+        compute_metrics=build_compute_metrics_fn(task_name))
+
+    tune_config = {
+        "per_device_eval_batch_size": 32,
+        "per_device_train_batch_size": tune.choice([16, 32, 64]),
```

Comment:
@krfricke I think you can actually remove `per_device_train_batch_size`,  `weight_decay`, and `learning_rate` from the config, so theres no need to specify it in both config and in PBT hyperparam_mutations. If these are not specified in config, the search space in PBT hyperparam_mutations is used instead.